app/pp/controllers/ChargeRefController.scala

```
@@ -56,9 +60,25 @@ class ChargeRefController @Inject() (
 
   def sendCardPaymentsNotification(): Action[ChargeRefNotificationRequest] = Action.async(parse.json[ChargeRefNotificationRequest]) { implicit request =>
     Logger.debug("sendCardPaymentsNotification")
-    processChargeRefNotificationRequest(request.body)
+    val ignoreSendChargeRef = checkSendChargeRefForTaxType(request.body.taxType)
+    if (ignoreSendChargeRef == false) {
+      processChargeRefNotificationRequest(request.body)
+    } else {
+      Logger.debug(s"Not sending des notification for ${request.body.taxType}, ignoreSendChargeRef was ${ignoreSendChargeRef}")
+      Future.successful(Ok)
+    }
+
   }
 
+  private def checkSendChargeRefForTaxType(taxType: TaxType) = {
+    import scala.collection.JavaConverters._
+
+    val ignoreList = configuration.underlying.getStringList("taxTypes.chargeref.ignore")
```

can the ignore list be created once-only?

---

Might be clearer to have:

def ignore(taxType: TaxType) = ignoreList.contains(taxType)

or even something like a better-named:

def shouldProcess(taxType: TaxType) = !ignoreList.contains(taxType)

rather than checking for false?

---

The reason to do it in config is that when we go live it will only cater for p800 but p800 will not be send to des ... so it allows a method to test the sending to des

---

ok re change to reverse boolean logic

---

Even if the list is configured it could still be created once, e.g. as a member variable:

    (lazy???) val ignoreList = configuration.underlying.getStringList("taxTypes.chargeref.ignore")
      .asScala.toList.map(m => TaxTypes.forCode(m).getOrElse(throw new RuntimeException(s"No TaxType for $m")))

don't think it needs to be lazy, presumably the configuration object is completely populated before being injected into the controller.

the main benefit is that will fail fast ifs there is a mismatch between code and config.

---

Use of a boolean flag makes all this redundant! Thank you.